File: 3_Outside_In/Soat.CleanCode.VideoStore.OutsideIn.TypeScript/src/pricer.ts
import { Category, Movie } from './movie';
import { Price }           from './price';

export function computePrice(movie: Movie) {
    if (movie.isNewRelease) {
        return Price.createNewRelease();
    }

    switch (movie.category) {
        case Category.Children:
            return Price.createChildren();

        case Category.Regular:
            return Price.createRegular();

        default:
            throw new RangeError(`movie.category (${movie.category}) is not a valid Category`);
    }
}

export type Pricer = typeof computePrice;

File: Soat.CleanCode.VideoStore.OutsideIn/Pricer.cs
using System;

namespace Soat.CleanCode.VideoStore.OutsideIn
{
    public class Pricer : IPricer
    {
        public IPrice GetPrice(Movie movie)
        {
            if (movie.ReleaseType == ReleaseType.New) {
                return Price.CreateNewRelease();
            }

            switch (movie.Category) {
                case MovieCategory.Children:
                    return Price.CreateChildren();

                case MovieCategory.Regular:
                    return Price.CreateRegular();

                default:
                    throw new ArgumentOutOfRangeException(nameof(movie), movie.Category, "Invalid movie category");
            }
        }
    }
}
